integration-cli/docker_api_containers_test.go

```diff
@@ -552,7 +552,7 @@ func (s *DockerSuite) TestContainerApiCommit(c *check.C) {
 	if err != nil {
 		c.Fatal(err)
 	}
-	if cmd != "{[/bin/sh -c touch /test]}" {
```

Comment:
ok I think I figured out what's going on.

What I think was happening was that `{{Config.Cmd}}` is a `stringutils.StrSlice`, which is defined as
```
type StrSlice struct {
    parts []string
}
```
Since the slice of strings are wrapped by a `struct` we get the extra `{}` around the cmd. Now, we don't see this in the JSON on the wire because we have a specialized JSON serializer func to hide it ( see https://github.com/docker/docker/blob/master/pkg/stringutils/strslice.go#L20 ).

However, I don't think the template.execute() stuff in the CLI code uses that specialized serializer, it just dumps the data as is - meaning with the struct wrapper.  If we do: `docker inspect -f "{{.Config.Cmd.Slice}}" ubuntu` and explicitly step into the struct to the `[]string` via the Slice() func then we'll get the correct results - meaning no `{}`. 

Net: While this PR changes the output of the template.execute() stuff, I think the change is correct. It does highlight a problem though, there could be other cases where using the `-f` option on inspect could produce the wrong output if the internal data type doesn't exactly match what's seen on the wire and a custom (de)serializer is used).  In the long run we should probably stop having the CLI use the same structs as the daemon and just use vanilla data types. We did a little bit of this when we created the api/types/types.go stuff but we may need to go further and do it for the runconfig structs too since those are pulled in.

Anyway, I feel better now that I think I know why this change happened.

Need to check one other change tho.....